src/components/study-container.js

```diff
@@ -0,0 +1,455 @@
+                    network.updateLines(values[0].lines);
+                    network.updateTwoWindingsTransformers(
+                        values[0].twoWindingsTransformers
+                    );
+                    network.updateThreeWindingsTransformers(
+                        values[0].threeWindingsTransformers
+                    );
+                    network.updateGenerators(values[0].generators);
+                    network.updateLoads(values[0].loads);
+                    network.updateBatteries(values[0].batteries);
+                    network.updateDanglingLines(values[0].danglingLines);
+                    network.updateLccConverterStations(
+                        values[0].lccConverterStations
+                    );
+                    network.updateVscConverterStations(
+                        values[0].vscConverterStations
+                    );
+                    network.updateHvdcLines(values[0].hvdcLines);
+                    network.updateShuntCompensators(
+                        values[0].shuntCompensators
+                    );
+                    network.updateStaticVarCompensators(
+                        values[0].staticVarCompensators
+                    );
+
+                    setUpdatedLines(values[0].lines);
+                })
+                .catch(function (error) {
+                    console.error(error.message);
+                    setNetworkLoadingFailMessage(error.message);
+                });
+            //.finally(() => setIsNetworkPending(false));
+            // Note: studyUuid don't change
+        },
+        [studyUuid, selectedNodeUuid, network]
+    );
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (
+                studyUpdatedForce.eventData.headers['updateType'] === 'loadflow'
+            ) {
+                //TODO reload data more intelligently
+                loadNetwork(true);
+            }
+        }
+        // Note: studyUuid, and loadNetwork don't change
+    }, [studyUpdatedForce, studyUuid, loadNetwork, dispatch]);
+
+    const runnable = useMemo(() => {
+        return {
+            LOADFLOW: intl.formatMessage({ id: 'LoadFlow' }),
+            SECURITY_ANALYSIS: intl.formatMessage({
+                id: 'SecurityAnalysis',
+            }),
+        };
+    }, [intl]);
+
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers) {
+            if (studyUpdatedForce.eventData.headers['updateType'] === 'study') {
+                // study partial update :
+                // loading equipments involved in the study modification and updating the network
+                const substationsIds =
+                    studyUpdatedForce.eventData.headers['substationsIds'];
+                const tmp = substationsIds.substring(
+                    1,
+                    substationsIds.length - 1
+                ); // removing square brackets
```

Comment:
what ? is it a complicated way to get the firs element of an array ?

Reply:
it seems so, the code was present in study-pane, so any fix would go to another PR

Reply:
ok